### ISSUE TITLE
[Feature]: Dealing with faulty (non-circular) averaging in SCADA

### ISSUE THREAD

**Issue by Bartdoekemeijer** [NREL/flasc#90]
### Description

FLASC currently assumes that the wind direction is the underlying SCADA has been correctly time-averaged, i.e., dealing with 360 deg wrapping for angular variables like wind direction and nacelle heading. However, this is often not the case when working with 10-minute averaged data from an operational asset. It would be great if FLASC could find these obvious outliers and mark them as faulty.

**Comment by paulf81** [comment]
That's a nice idea, just wanted to jot down some ideas while fresh:

- Flag changes in direction above some threshold (90 deg?)
- Threshold based approach: toggling between <20 | > 340 and >160 & < 200
- some sort of expectation comparison, like predict the value of a point as being the midpoint of the point before and after it, and a point that is off by more than 100 deg is flagged

**Comment by Bartdoekemeijer** [comment]
Yeah good ideas! Maybe the expectation comparison can leverage the neighbouring turbines? E.g., if the median change in wind direction of all turbines is 10 deg, but there are one or two turbines reporting 150 degrees, then those are probably outliers.

**Comment by paulf81** [comment]
That also makes sense to me, could be a threshold like, can't exceed some X times the median change?

**Comment by aclerc** [comment]
This is a feature in wind-up. Happy to discuss further if it's of interest
